fix(linter): Fix docs for `consistent-type-specifier-style` and `consistent-type-definitions` rules.

connorshea opened the pull request:
Both of these take a string config, and the docs were set up as a configuration object.

Also refactor a few more rules to use the DefaultRuleConfig pattern.

camc314 commented:
### Merge activity

* **Nov 28, 10:41 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 28, 10:43 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

connorshea committed:
fix(linter): Fix docs for `consistent-type-specifier-style` and `consistent-type-definitions` rules. (#16253)

Both of these take a string config, and the docs were set up as a configuration object.

Also refactor a few more rules to use the DefaultRuleConfig pattern.
